File: src/main/scala/hdzi/swordsman/scala/_07_Collections.scala
package hdzi.swordsman.scala

import scala.collection.mutable
import scala.collection.mutable.ArrayBuffer

/**
  * Created by taojinhou on 2018/8/22.
  */
object _07_Collections {
  /**
    * 数组
    */
  val arr0 = new Array[Int](10) // size
  val arr1 = Array(1, 2, 3, 4, 5, 6, 7, 8, 9) // 使用apply方法获取一个定义好的array

  /**
    * 变长数组
    */
  val marr = new ArrayBuffer[Int]()

  /**
    * 不可变列表
    */
  val list = List(1, 2, 3, 4, 5, 6)
  val list2 = 1 :: 2 :: 3 :: 4 :: Nil

  /**
    * 可变列表
    */
  val mlist = mutable.MutableList(1, 2, 3, 4)

  /**
    * map
    */
  val map = Map(1 -> "one", 2 -> "two")

  /**
    * 区间
    */
  val range0 = 1 to 10 // [1, 10]
  var range1 = 1 until 10 // [1, 10)
  var range2 = 10 to 1 by -1 // 10 到 1, 步长 -1

  /**
    * 元组
    */
  val tuple1 = (1, "one")
  val tuple2 = 2 -> "two" // -> 只适用两元元组
  val one = tuple1._1 // 元组从1开始计数
  val (a, b, c) = (1, "2", 3.0)

}

File: src/main/kotlin/hdzi/swordsman/kotlin/_07_Collections.kt
package hdzi.swordsman.kotlin

/**
 * Created by taojinhou on 2018/8/20.
 */

/**
 * 区间
 */
val range1 = 1..100 // 1到100
val range2 = 100 downTo 1 // 100 到 1
val range3 = 1..100 step 2 // 1到100,步长2
val range4 = 1 until 100 // 1到99

/**
 * in 检查成员
 */
val _in = 2 in range1
val _not_in = -1 !in range1

/**
 *  声明集合:kotlin的集合声明基本依靠工具函数
 */
// 数组
val arrInt = intArrayOf(1, 2, 3) // 内建类型专用(int,long,boolean,float等值类型)
val arrString = arrayOf("A", "B", "C") // 通用
// list
val listInt = listOf(1, 2, 3)
// map
val mapIntString = mapOf(1 to "one", 2 to "two") // 用 to 方法生成Pair<K, V>
// 可变集合 mutable*()
val mlist = mutableListOf(1, 2, 3)

/**
 * 存取集合
 */
fun changeCollection() {
    // 数组
    var ele = arrInt[0] // get方法
    arrInt[0] = 1 // set方法
    // 集合
    ele = listInt[0]
    mlist[0] = 0
}